http/client_pool.lua

```diff
@@ -0,0 +1,142 @@
+local cs = require "cqueues.socket"
+
+local function reuse_connection(candidate, connect_options)
+	-- Assume family/host/port/path already checked
+
+	if candidate.socket == nil then
+		return false
+	end
+
+	if connect_options.v6only then
+		-- TODO
+		return false
+	end
+
+	local bind = connect_options.bind
+	if bind then
+		-- TODO: Use :localname()
+		return false
+	end
+
+	local version = connect_options.version
+	if version and version ~= candidate.version then
+		return false
+	end
+
+	if candidate.version < 2 then
+		-- Check if connection already in use (avoid pipelining)
```

Comment:
TODO: check pipeline depth instead.
Also, might want to find shortest pipeline in the pool?